insights/parsers/package_provides_java.py

```diff
@@ -0,0 +1,49 @@
+"""
+PackageProvidesJava - Command
+=============================
+
+This parser reads the output from pre_command
+``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; \
+do echo $jp `readlink -e $jp | xargs rpm -qf`; done``
+and coverts it into a dict that links running_java_command with its package name.
+The reason why using above pre_command is that we need to record the links between running_java_command
+and package which provides the java command. In ``ps aux`` output, we can only get what java command starts a
+java application, instead of java package.
+Through this way, when there is jdk bug, we can detect whether a running java application will be affected.
+
+
+Typical contents of the pre_command::
+
+    /usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64
+
+Parsed result::
+
+    {"command": "/usr/lib/jvm/jre/bin/java", "package": "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"}
+
+Examples:
+
+    >>> command_package = shared[PackageProvidesJava]
+    >>> command_package.data
```

Comment:
Since it's already mixin with the `LegacyItemAccess`, better avoid using the `.data` in the example.

Reply:
Have updated according to this comment.